source/common/access_log/access_log_formatter.h

```diff
@@ -95,14 +95,33 @@ class FormatterImpl : public Formatter {
                      const StreamInfo::StreamInfo& stream_info) const override;
 
 private:
-  std::vector<FormatterPtr> formatters_;
+  std::vector<FormatterProviderPtr> providers_;
+};
+
+class JsonFormatterImpl : public Formatter {
+public:
+  JsonFormatterImpl(const std::map<const std::string, const std::string>& format_mapping);
+
+  // Formatter::format
+  std::string format(const Http::HeaderMap& request_headers,
+                     const Http::HeaderMap& response_headers,
+                     const Http::HeaderMap& response_trailers,
+                     const StreamInfo::StreamInfo& stream_info) const override;
+
+  std::unordered_map<std::string, std::string>
+  toMap(const Http::HeaderMap& request_headers, const Http::HeaderMap& response_headers,
+        const Http::HeaderMap& response_trailers, const StreamInfo::StreamInfo& stream_info) const;
+
+private:
+  std::vector<FormatterProviderPtr> providers_;
+  std::map<const std::string, Envoy::AccessLog::FormatterProviderPtr> json_output_format_;
```

Comment:
Can you add a comment everywhere there is `std::map` indicating why order matters (or switch to `unordered_map`)?

Reply:
Yeah. Will do

Reply:
Order didn't matter here either, changed to `unordered`